File: MyMelvor/mymelvor.client/src/classes/InventoryItemClass.js
import ItemArray from "../classes/ItemArray"
import ItemClass from "./ItemClass"
import ItemType from "./ItemType"

export default class InventoryItemClass extends ItemClass {
    Count = 0

    constructor(id, count) {
        super(id, "", "", ItemType.Wood)
        this.Count = count

        var foundItem = ItemArray.find(i => i.Id == id)
        if (foundItem !== undefined) {
            this.Name = foundItem.Name
            this.Description = foundItem.Description
            this.Type = foundItem.Type
        }
    }
}

File: MyMelvorBlazor/MyMelvorBlazor/Models/InventoryItemClass.cs
namespace MyMelvorBlazor.Models
{
	public class InventoryItemClass
	{
		public ItemId? Id { get; set; } = null;
		public string Name { get; set; } = "";
		public string Description { get; set; } = "";
		public int Count { get; set; } = 0;

		public InventoryItemClass(ItemId id, int count)
		{
			Id = id;
			Count = count;

			var foundItem = ItemArray.Items.First(i => i.Id == id);
			if (foundItem != null)
			{
				Name = foundItem.Name;
				Description = foundItem.Description;
			}
		}
	}
}